docker/buildkite/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.13.3
+FROM golang:1.17
```

Comment:
do you mind if we pull these out into separate diffs too? sorry to be annoying

Reply:
Yes. https://github.com/uber/cadence/pull/4889 I'm fixing some test failures.

Reply:
it's merged.